airbyte-webapp/src/core/request/CommonRequestError.ts

```diff
@@ -1,17 +1,16 @@
 export class CommonRequestError extends Error {
   __type = "common.error";
-  response: Response;
   // TODO: Add better error hierarchy
   _status?: number;
 
-  constructor(response: Response, msg?: string) {
-    super(response.statusText);
+  constructor(protected response: Response | undefined, msg?: string) {
+    super(response?.statusText);
     this.response = response;
     this.message = msg ?? "common.error";
   }
 
   get status(): number {
-    return this._status || this.response.status;
+    return (this._status || this.response?.status) ?? 400;
```

Comment:
What changed that led to adding a `400` as a fallback?

Reply:
It's because `this.response` can be undefined based on the new types.
It's only used in one place: `if (error.status === 422 || error.status === 404) {`
So this is safe returning undefined which I think is the better option. Not faking a 400.